File: csharp/create-complaint.cs
string prodUrl = "<REPLACE_WITH_PROD_URL>";
string token = "<REPLACE_WITH_YOUR_API_KEY>";

var client = new RestClient($"{prodUrl}/api/v1/complaints");
client.Timeout = -1;

var request = new RestRequest(Method.POST);
request.AddHeader("Authorization", $"Bearer {token}");
request.AddHeader("Content-Type", "application/json");

var requestBody = new {
    complaintTypeId = 1,
    customerBankAccountNumber = "1234567890",
    customerName = "Nom du client",
    customerPhoneNumber = "2250555607018",
    customerEmail = "email@example.com",
    description = "Description de la réclamation",
    dynamicInfo = "{\"name\": \"my name\"}" // Objet stringifié
};

request.AddJsonBody(requestBody);

IRestResponse response = client.Execute(request);
Console.WriteLine(response.Content);

File: java/create-complaint.java
import okhttp3.*;

import java.io.IOException;

public class Main {
    public static void main(String[] args) throws IOException {
        OkHttpClient client = new OkHttpClient().newBuilder().build();

        MediaType mediaType = MediaType.parse("application/json");
        RequestBody body = RequestBody.create(mediaType, "{\n" +
                "    \"complaintTypeId\": 1,\n" +
                "    \"customerBankAccountNumber\": \"1234567890\",\n" +
                "    \"customerName\": \"Nom du client\",\n" +
                "    \"customerPhoneNumber\": \"2250555607018\",\n" +
                "    \"customerEmail\": \"email@example.com\",\n" +
                "    \"description\": \"Description de la réclamation\",\n" +
                "    \"dynamicInfo\": \"{\\\"name\\\": \\\"my name\\\"}\" // Objet stringifié\n" +
                "}");

        Request request = new Request.Builder()
                .url("<REPLACE_WITH_PROD_URL>/api/v1/complaints")
                .method("POST", body)
                .addHeader("Authorization", "Bearer <REPLACE_WITH_YOUR_API_KEY>")
                .addHeader("Content-Type", "application/json")
                .build();

        Response response = client.newCall(request).execute();
        String responseBody = response.body().string();
        System.out.println(responseBody);
    }
}
